### PR TITLE
Using directory as prefix for S3

### PR DESCRIPTION
**Pull request by linzhp**
## What is the problem I am trying to address?

Listing objects on S3 with prefixes like `github.com/golang/mock/@v/v1.4.4.` with return empty list, because prefixes are directories on S3 (https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-prefixes.html).

## How is the fix applied?

Taking the directory name from the path returned by `config.PackageVersionedName(module, version, "")`, so  it  can return all files for the module. Because of this, all files from all versions will be returned and filtered later.

### CODE REVIEW

File: pkg/storage/s3/checker.go [hunk 1]
```diff
@@ -23,7 +24,7 @@ func (s *Storage) Exists(ctx context.Context, module, version string) (bool, err
 
 	lsParams := &s3.ListObjectsInput{
 		Bucket: aws.String(s.bucket),
-		Prefix: aws.String(config.PackageVersionedName(module, version, "")),
+		Prefix: aws.String(path.Dir(config.PackageVersionedName(module, version, ""))),
```
Comment (marwan-at-work) [inline pkg/storage/s3/checker.go:27]: I think you can just use `config.FmtModVer` instead of taking path.Dir(config.PackageVersionedName(...)).

**Review by marwan-at-work — COMMENTED** [review]
Thanks for the fix. I'm surprised this hasn't come up before. 

I think we could really use some integration testing here but unfortunately I don't have an AWS account. 

cc: @arschles

File: pkg/storage/s3/checker.go (hunk 1 above)
Comment (linzhp) [reply pkg/storage/s3/checker.go:27]: `config.FmtModVer` would return something like `github.com/golang/mock@1.4.4`, but we need `github.com/golang/mock/@v`

**Comment by linzhp** [comment]
@marwan-at-work Can you take another look?

File: pkg/storage/s3/checker.go (hunk 1 above)
Comment (marwan-at-work) [reply pkg/storage/s3/checker.go:27]: @linzhp in that case adding a version then removing just to use existing helpers seems a little unnecessary. What if we just did this: `aws.String(fmt.Sprintf("%s/@v", module))`

**Comment by marwan-at-work** [comment]
@linzhp just a nit comment and looks good 👍🏼

**Review by marwan-at-work — APPROVED** [review]

**Pull request merged**